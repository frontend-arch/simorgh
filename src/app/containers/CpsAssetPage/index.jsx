import React from 'react';
import compose from 'ramda/src/compose';
import CpsAssetPageMain from '../CpsAssetPageMain';
import withContexts from '../PageHandlers/withContexts';
import withPageWrapper from '../PageHandlers/withPageWrapper';
import withError from '../PageHandlers/withError';
import withLoading from '../PageHandlers/withLoading';
import withData from '../PageHandlers/withData';

const CpsContainer = props => {
  return <CpsAssetPageMain {...props} />;
};

const EnhancedCpsContainer = compose(
  withContexts,
  withPageWrapper,
  withLoading,
  withError,
  withData,
)(CpsContainer);

export default EnhancedCpsContainer;
