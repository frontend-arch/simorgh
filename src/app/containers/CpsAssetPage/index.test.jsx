import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import CpsAssetPage from '.';
import pidginPageData from '#data/pidgin/cpsAssets/tori-49450859';

const cpsAssetScaffoldProps = {
  isAmp: false,
  pageType: 'MAP',
  service: 'pidgin',
  pathname: '/pidgin/tori-49450859',
  match: {
    params: {
      assetUri: 'tori-49450859',
      pageData: pidginPageData,
    },
  },
  status: 200,
};

jest.mock('../PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../PageHandlers/withError', () => Component => {
  const ErrorContainer = props => (
    <div id="ErrorContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});

jest.mock('../PageHandlers/withData', () => Component => {
  const DataContainer = props => (
    <div id="DataContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../CpsAssetPageMain', () => {
  const CpsAssetPageMain = () => <div>CpsAssetPageMain</div>;

  return CpsAssetPageMain;
});

describe('CpsAssetPage', () => {
  describe('snapshots', () => {
    shouldMatchSnapshot(
      'should match scaffold snapshot',
      <CpsAssetPage {...cpsAssetScaffoldProps} />,
    );
  });
});
