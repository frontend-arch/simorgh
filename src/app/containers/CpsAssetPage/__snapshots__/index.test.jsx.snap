// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpsAssetPage snapshots should match scaffold snapshot 1`] = `
<div
  id="PageWrapperContainer"
>
  <div
    id="LoadingContainer"
  >
    <div
      id="ErrorContainer"
    >
      <div
        id="DataContainer"
      >
        <div>
          CpsAssetPageMain
        </div>
      </div>
    </div>
  </div>
</div>
`;
